lib/pgBackRest/Storage/S3/Driver.pm

```diff
@@ -0,0 +1,512 @@
+sub pathSync
+{
+    my $self = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $strPath,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '->pathSync', \@_,
+            {name => 'strPath', trace => true},
+        );
+
+    # Return from function and log return values if any
+    return logDebugReturn($strOperation);
+}
+
+####################################################################################################################################
+# exists - check if a file exists
+####################################################################################################################################
+sub exists
+{
+    my $self = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $strFile,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '->exists', \@_,
+            {name => 'strFile', trace => true},
+        );
+
+    # Does the path/file exist?
+    my $bExists = defined($self->manifest($strFile, {bRecurse => false, bPath => false})->{basename($strFile)}) ? true : false;
+
+    # Return from function and log return values if any
+    return logDebugReturn
+    (
+        $strOperation,
+        {name => 'bExists', value => $bExists, trace => true}
+    );
+}
+
+####################################################################################################################################
+# pathExists
+####################################################################################################################################
+sub pathExists
+{
+    my $self = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $strPath,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '->pathExists', \@_,
+            {name => 'strPath', trace => true},
+        );
+
+    my $bExists = true;
+
+    # Only check if path <> /
+    if ($strPath ne qw{/})
+    {
+        # Does the path exist?
+        my $rhInfo = $self->manifest(dirname($strPath), {bRecurse => false, bPath => true})->{basename($strPath)};
+        $bExists = defined($rhInfo) && $rhInfo->{type} eq 'd' ? true : false;
+    }
+
+    # Return from function and log return values if any
+    return logDebugReturn
+    (
+        $strOperation,
+        {name => 'bExists', value => $bExists, trace => true}
+    );
+}
+
+####################################################################################################################################
+# info - get information about a file
+####################################################################################################################################
+sub info
+{
+    my $self = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $strFile,
+    ) =
```

Comment:
That's an important option when building manifests from the database directory but not useful in a repo where files are not supposed to disappear suddenly.  We could include it for the sake on completeness but it would probably never be used.

We could also handle cases like this with capabilities - i.e. ignore missing on info is not allowed, but I don't think it's worth the extra code.  Or maybe that's a refinement.